astra/src/main/java/com/slack/astra/logstore/search/SearchResultAggregatorImpl.java

```diff
@@ -55,8 +65,16 @@ public SearchResult<T> aggregate(List<SearchResult<T>> searchResults, boolean fi
       // The last aggregation should be indicated using the final aggregation boolean. This performs
       // some final pass "destructive" actions, such as applying min doc count or extended bounds.
       if (finalAggregation) {
-        pipelineTree =
-            OpenSearchAdapter.getAggregationBuilder(searchQuery.aggBuilder).buildPipelineTree();
+        if (searchQuery.aggregatorFactoriesBuilder != null
+            && this.useOpenSearchAggregationParsing) {
+          Collection<AggregationBuilder> aggregationBuilders =
+              searchQuery.aggregatorFactoriesBuilder.getAggregatorFactories();
+          pipelineTree = aggregationBuilders.stream().toList().getFirst().buildPipelineTree();
```

Comment:
This seems a bit odd - going a stream, to list, and then get first. Is there a cleaner way to easily do this?